File: src/main/java/cz/kamenitxan/df3d/pages/IndexPage.kt
package cz.kamenitxan.df3d.pages

import cz.kamenitxan.df3d.model.Product
import cz.kamenitxan.jakon.core.customPages.CustomPage
import cz.kamenitxan.jakon.core.model.Dao.DBHelper
import cz.kamenitxan.jakon.core.template.TemplateUtils
import cz.kamenitxan.jakon.webui.ObjectSettings
import java.util.HashMap

class IndexPage : CustomPage() {

	override fun render() {
		val engine = TemplateUtils.getEngine()
		val products = DBHelper.getSession().createCriteria(Product::class.java).list()
		val model = HashMap<String, Any>()
		model.put("products", products)
		engine.render("index", "index", model)
	}

	override fun objectSettings(): ObjectSettings? {
		return null
	}

}

File: src/main/java/cz/kamenitxan/df3d/pages/IndexPage.scala
package cz.kamenitxan.df3d.pages

import java.util

import cz.kamenitxan.df3d.model.ProductService
import cz.kamenitxan.jakon.core.configuration.Settings
import cz.kamenitxan.jakon.core.customPages.AbstractCustomPage
import cz.kamenitxan.jakon.core.model.Dao.DBHelper
import cz.kamenitxan.jakon.core.template.TemplateUtils
import cz.kamenitxan.jakon.webui.ObjectSettings

import scala.collection.JavaConverters._

class IndexPage extends AbstractCustomPage {

	override protected def generate(): Unit = {
		val engine = TemplateUtils.getEngine

		implicit val conn = DBHelper.getConnection
		val products = ProductService.getAll
		conn.close()

		Settings.setDefaultLocale("cs_CZ")
		val model = new util.HashMap[String, AnyRef]()
		model.put("products", products.asJava)
		model.put("lang", "cs")
		engine.render("index", "index", model)

		Settings.setDefaultLocale("la_VA")
		val latinModel = new util.HashMap[String, AnyRef]()
		latinModel.put("products", products.asJava)
		latinModel.put("lang", "la")
		engine.render("index", "latin", latinModel)
	}

	override val objectSettings: ObjectSettings = null
}
